testsuite/tests/kuadrant/authorino/identity/api_key/conftest.py

```diff
@@ -5,34 +5,32 @@
 
 
 @pytest.fixture(scope="module")
-def invalid_label_selector():
-    """Label for API key secret that is different from the one specified in AuthConfig"""
-    return "invalid_api_label"
-
-
-@pytest.fixture(scope="module")
-def api_key(create_api_key, module_label):
+def api_key_secret(create_api_key, module_label):
     """Creates API key Secret"""
     api_key = "api_key_value"
-    create_api_key("api-key", module_label, api_key)
-    return api_key
+    return create_api_key("api-key", module_label, api_key)
 
 
 @pytest.fixture(scope="module")
-def auth(api_key):
+def auth(api_key_secret):
     """Valid API Key Auth"""
-    return HeaderApiKeyAuth(api_key)
+    return HeaderApiKeyAuth(str(api_key_secret))
+
+
+@pytest.fixture(scope="module")
+def invalid_label_selector():
+    """Label for API key secret that is different from the one specified in AuthConfig"""
+    return "invalid_api_label"
 
 
 @pytest.fixture(scope="module")
-def invalid_api_key(create_api_key, invalid_label_selector):
+def invalid_api_key_secret(create_api_key, invalid_label_selector):
     """Creates API key Secret with label that does not match any of the labelSelectors defined by AuthConfig"""
     api_key = "invalid_api_key"
-    create_api_key("invalid-api-key", invalid_label_selector, api_key)
-    return api_key
+    return create_api_key("invalid-api-key", invalid_label_selector, api_key)
 
 
 @pytest.fixture(scope="module")
-def invalid_auth(invalid_api_key):
-    """Invalid key Auth"""
-    return HeaderApiKeyAuth(invalid_api_key)
+def invalid_auth(invalid_api_key_secret):
+    """Invalid API key Auth"""
+    return HeaderApiKeyAuth(str(invalid_api_key_secret))
```

Comment:
You could update the Class to take either ApiKey or string

Reply:
Yeah I did that when I was reviewing my changes :D